bee-network/src/conns/dial.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{errors::Error, manager::LISTEN_ADDRESSES, Origin};
+
+use crate::{
+    interaction::events::InternalEventSender,
+    peers::{BannedAddrList, BannedPeerList, PeerInfo, PeerList, PeerRelation, PeerState},
+    transport::build_transport,
+    Multiaddr, PeerId, ShortId,
+};
+
+use log::*;
+
+use libp2p::{identity, Transport};
+
+pub async fn dial_peer(
+    peer_id: &PeerId,
+    local_keys: &identity::Keypair,
+    internal_event_sender: &InternalEventSender,
+    peers: &PeerList,
+    banned_addrs: &BannedAddrList,
+    banned_peers: &BannedPeerList,
+) -> Result<(), Error> {
+    // Prevent dialing oneself.
+    if peer_id == &local_keys.public().into_peer_id() {
+        return Err(Error::DialedSelf(peer_id.short()));
+    }
+
+    // Prevent duplicate connections.
+    if let Ok(connected) = peers.is(peer_id, |_, state| state.is_connected()).await {
```

Comment:
it does, but the compiler will easily detect that as well and remove the unnecessary nested branch, but you tell me :stuck_out_tongue: 
I wrote it that way because I personally think `If let Ok(connected) ...` is more readable than `If let Ok(true) ....`.

Reply:
anyway, I think reading the full line is still pretty readable, so I make the change :smile: